contribs/drt-extensions/src/main/java/org/matsim/contrib/drt/extension/estimator/BasicDrtEstimator.java

```diff
@@ -0,0 +1,174 @@
+package org.matsim.contrib.drt.extension.estimator;
+
+import org.apache.commons.math3.stat.descriptive.SummaryStatistics;
+import org.apache.commons.math3.stat.regression.RegressionResults;
+import org.apache.commons.math3.stat.regression.SimpleRegression;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.matsim.api.core.v01.events.PersonMoneyEvent;
+import org.matsim.contrib.drt.analysis.DrtEventSequenceCollector;
+import org.matsim.contrib.drt.extension.estimator.run.DrtEstimatorConfigGroup;
+import org.matsim.contrib.drt.fare.DrtFareParams;
+import org.matsim.contrib.drt.routing.DrtRoute;
+import org.matsim.contrib.drt.run.DrtConfigGroup;
+import org.matsim.contrib.drt.speedup.DrtSpeedUp;
+import org.matsim.contrib.drt.speedup.DrtSpeedUpParams;
+import org.matsim.core.controler.events.IterationEndsEvent;
+import org.matsim.core.controler.listener.IterationEndsListener;
+import org.matsim.core.utils.misc.OptionalTime;
+
+import java.util.SplittableRandom;
+
+/**
+ * Estimates drt trips based only daily averages. No spatial or temporal differentiation is taken into account for the estimate.
+ * This estimator is suited for small scenarios with few vehicles and trips and consequently few data points.
+ */
+public class BasicDrtEstimator implements DrtEstimator, IterationEndsListener {
+
+	private static final Logger log = LogManager.getLogger(BasicDrtEstimator.class);
+
+	private final DrtEventSequenceCollector collector;
+	private final DrtEstimatorConfigGroup config;
+	private final DrtConfigGroup drtConfig;
+
+	private final DrtSpeedUpParams speedUpParams;
+	private final SplittableRandom rnd = new SplittableRandom();
+	/**
+	 * Currently valid estimates.
+	 */
+	private GlobalEstimate currentEst;
+	private RegressionResults fare;
+
+	public BasicDrtEstimator(DrtEventSequenceCollector collector, DrtEstimatorConfigGroup config,
+							 DrtConfigGroup drtConfig) {
+		//zones = injector.getModal(DrtZonalSystem.class);
+		this.collector = collector;
+		this.config = config;
+		this.drtConfig = drtConfig;
+		this.speedUpParams = this.drtConfig.getDrtSpeedUpParams().isPresent() ? this.drtConfig.getDrtSpeedUpParams().get() : null;
```

Comment:
This could be replaced with a shorter version: `this.drtConfig.getDrtSpeedUpParams().orElse(null)`.

We could also have `Optional<DrtSpeedUpParams>` (despite IntelliJ suggesting that this is a bad practice), then line 55-56 would need to be adjusted accordingly.

Reply:
Or you can always simply call `drtConfig.getDrtSpeedUpParams()` (without storing it) - same way as the drt fare params are obtained in `estimate()`.